src/pepr/operator/controllers/keycloak/clients/dynamic-client.ts

```diff
@@ -0,0 +1,52 @@
+/**
+ * Copyright 2025 Defense Unicorns
+ * SPDX-License-Identifier: AGPL-3.0-or-later OR LicenseRef-Defense-Unicorns-Commercial
+ */
+
+import { Client } from "../types";
+import {
+  credentialsCreateOrUpdate,
+  credentialsDelete,
+  credentialsGetAccessToken,
+} from "./client-credentials";
+import { dynamicCreateOrUpdate, dynamicDelete } from "./dynamic-client-registration";
+import { log } from "./common";
+
+export async function getStrategy() {
+  const strategy = process.env.PEPR_KEYCLOAK_CLIENT_STRATEGY || "auto";
+  if (strategy === "client_credentials") {
+    log.debug("Using Client Credentials strategy");
+    return "client_credentials";
+  } else if (strategy === "auto") {
+    try {
+      log.debug("Probing Client Credentials strategy");
+      await credentialsGetAccessToken();
+      log.debug("Using Client Credentials strategy");
+      return "client_credentials";
+    } catch {
+      log.warn("Cannot use Client Credentials, falling back to dynamic registration");
+      return "dynamic";
+    }
+  } else {
+    log.warn(
+      `Invalid ${process.env.PEPR_KEYCLOAK_CLIENT_STRATEGY} parameter value, falling back to dynamic registration`,
+    );
+    return "dynamic";
+  }
+}
```

Comment:
Bit of a nit but it would be nice to use something "typed" here like an enum rather than just a string. We can keep it really simple, for example:
```ts
enum ClientStrategy {
  AUTO = "auto",
  CLIENT_CREDENTIALS = "client-credentials",
  DYNAMIC = "dynamic",
}
```

This just helps make sure we have better type checking/protection in the code, rather than relying on string matching.

Reply:
Fixed